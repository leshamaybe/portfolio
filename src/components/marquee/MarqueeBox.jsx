import React from 'react';
import Marquee from 'react-fast-marquee';
import s from './marquee.module.scss';
import useWindowSize from '@/hooks/useWindowSize';

const MarqueeBox = () => {
    const size = useWindowSize();

    return (
        <>
            {size.width > 750 && (
                <Marquee gradientColor="[255, 255, 255]" speed="150" className={s.marquee}>
                    <h1>AVAILABLE FOR WORK</h1>
                </Marquee>
            )}
        </>
    );
};

export default MarqueeBox;
