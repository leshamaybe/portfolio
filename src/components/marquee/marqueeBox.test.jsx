import MarqueeBox from './MarqueeBox';
import { render } from '@testing-library/react';

describe('MarqueeBox', () => {
    it('MarqueeBox renders correctly', () => {
        const { container } = render(<MarqueeBox />);
        expect(container).toMatchSnapshot();
    });
});
