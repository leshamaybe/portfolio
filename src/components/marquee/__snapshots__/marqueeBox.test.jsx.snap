// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarqueeBox MarqueeBox renders correctly 1`] = `
<div>
  <div
    class="max-w-120+50 bg-light-grey h-25+2 rounded-[20px] duration-[0.25s] cursor-pointer border-[1.5px] border-transparent transition-all hover:scale-105 hover:border-[1.5px] hover:border-dark-grey marquee-container"
    style="--pause-on-hover: running; --pause-on-click: running;"
  >
    <div
      class="overlay"
      style="--gradient-color: rgba([, 2, 5, 1), rgba([, 2, 5, 0); --gradient-width: 200px;"
    />
    <div
      class="marquee"
      style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite;"
    >
      <h1
        class="flex items-center text-12+1 text-normal before:marquee-circle"
      >
        AVAILABLE FOR WORK
      </h1>
    </div>
    <div
      aria-hidden="true"
      class="marquee"
      style="--play: running; --direction: normal; --duration: 0s; --delay: 0s; --iteration-count: infinite;"
    >
      <h1
        class="flex items-center text-12+1 text-normal before:marquee-circle"
      >
        AVAILABLE FOR WORK
      </h1>
    </div>
  </div>
</div>
`;
